### DISCUSSION TITLE
Has util.ajax() been removed??

### DISCUSSION THREAD

**Discussion by BernhardBaumrock** [uikit/uikit#4784] [General]
I'm trying to use ajax as described here and as I've always done: https://github.com/uikit/uikit-site/blob/feature/js-utils/docs/pages/javascript-utilities.md#ajax-1

> Uncaught TypeError: util.ajax is not a function

Has ajax method been removed? Or why is it not working any more on my end? Any ideas?

Thx!

**Comment by aarongerig** [comment]
@BernhardBaumrock It's been removed with version 3.16.0 (see https://getuikit.com/changelog).

**Reply by BernhardBaumrock** [reply]
Thx!

**Reply by viocassel** [reply]
> Thx!

Use `fetch()` 😉

**Comment by BernhardBaumrock** [comment]
@janschoenherr could someone please update the "docs" at https://github.com/uikit/uikit-site/blob/feature/js-utils/docs/pages/javascript-utilities.md ?

Why is that file not part of the official docs? I've been using UIkit for years without knowing anything about it's great javascript capabilities until someone in another forum pointed me to that file ...

It would be great to at least mention the existence of it somewhere in the docs.

**Reply by aarongerig** [reply]
@BernhardBaumrock I started to create that JS utility documentation a couple of years back, but never got around to finish it, hence the reason it is not included on the docs website and stays only available on GitHub. The source code of the docs is open to the whole UIkit open source community, but it simply didn't receive enough contributions for it to be published on the website. 🤷🏻‍♂️

Edit: Btw, you yourself could update the docs too by creating a PR. 😉

**Reply by BernhardBaumrock** [reply]
@aarongerig It was totally knew to me that these "docs" were done by a volunteer. So thx for your work on this. I was a huge help for me when developing websites!

Also thx for educating me about PRs -  the problem here is that if I were experienced enough for reading uikits source files and understanding them in a way that I could write docs for them, then I'd not need the docs. I need them because I don't understand uikits raw source files and I need help and easy to understand examples.

For example I don't understand this:
```js
    function each(obj, cb) {
      for (const key in obj) {
        if (false === cb(obj[key], key)) {
          return false;
        }
      }
      return true;
    }
```

But I easily understand this:
```js
var array = ['Apple', 'Banana', 'Kiwi', 'Mango'];

util.each(array, function (item) {
  console.log(item);
});
```

I think uikit's JS tools are great and I simply don't understand why the uikit company does not put more effort into promoting the strenghts of its own product and just leave a hidden treasure lying around. It would be a very low hanging fruit in my opinion.
